### PR TITLE
Update statickcheck for Go 1.18

### PR DESCRIPTION
**Pull request by adrianludwin**
Tested: static check fails on Go 1.18 without this change, passes with
it.

### CODE REVIEW

**Comment by erikgb** [comment]
@adrianludwin This does not seem to work using Go 1.17, and I do not think we should require 1.18 before K8s is fully migrated to 1.18. I get the same error as the CI when trying to build your branch:
````
D:~/projects/github/hierarchical-namespaces $ go version
go version go1.17.8 linux/amd64
D:~/projects/github/hierarchical-namespaces $ make
go build -o /home/erikbo/projects/github/hierarchical-namespaces/bin/controller-gen sigs.k8s.io/controller-tools/cmd/controller-gen
/home/erikbo/projects/github/hierarchical-namespaces/bin/controller-gen object:headerFile=./hack/boilerplate.go.txt paths=./api/...
gofmt -l -w ./api ./cmd ./internal
go vet ./api/... ./cmd/... ./internal/...
go build -o /home/erikbo/projects/github/hierarchical-namespaces/bin/staticcheck honnef.co/go/tools/cmd/staticcheck
/home/erikbo/projects/github/hierarchical-namespaces/bin/staticcheck ./api/... ./cmd/... ./internal/...
internal/kubectl/configset.go:81:2: this linter directive didn't match anything; should it be removed? (staticcheck)
make: *** [Makefile:206: staticcheck] Error 1
D:~/projects/github/hierarchical-namespaces $ 
````

File: internal/kubectl/configset.go [hunk 1]
```diff
@@ -78,6 +78,7 @@ var setResourceCmd = &cobra.Command{
 // normalizeModeString corrects format of input Synchronization Mode string
 func normalizeModeString(modeStr string) string {
 	low := strings.ToLower(modeStr)
+	//lint:ignore SA1019 we should fix this but not today
 	return strings.Title(low)
```
Comment (erikgb) [inline internal/kubectl/configset.go:82]: We could consider fixing this staticcheck issue now - just to fix the build (using Go 1.17)

```suggestion
	return cases.Title(language.English).String(low)
```

Comment (adrianludwin) [reply internal/kubectl/configset.go:82]: Ok, will do

**Pull request merged**